packages/common/src/api/tan-query/useUpdateTrack.ts

```diff
@@ -1,83 +1,171 @@
   const dispatch = useDispatch()
-  const store = useStore()
-  const { mutate: deleteTrack } = useDeleteTrack()
+  const { data: currentUser } = useCurrentUser()
+  const { mutate: updateStems } = useUpdateStems()
+  const { data: userBank } = useGetOrCreateUserBank(
+    currentUser?.erc_wallet ?? currentUser?.wallet
+  )
+  const { isEnabled: isUsdcPurchaseEnabled } = useFeatureFlag(
+    FeatureFlags.USDC_PURCHASES
+  )
 
   return useMutation({
     mutationFn: async ({
       trackId,
-      userId,
       metadata,
       coverArtFile
     }: UpdateTrackParams) => {
+      if (!trackId) {
+        throw new Error('Track ID is required')
+      }
+
       const sdk = await audiusSdk()
 
       const previousMetadata = queryClient.getQueryData<Track>([
         QUERY_KEYS.track,
         trackId
       ])
+
+      // Create a mutable copy of metadata
+      let updatedMetadata = {
+        ...metadata
+      } as Track & TrackMetadataForUpload
+
+      // Apply squashNewLines to description
+      if (updatedMetadata.description) {
+        updatedMetadata.description = squashNewLines(
+          updatedMetadata.description
+        )
+      }
+
+      // Handle publishing state for unlisted to listed transition
+      if (
+        previousMetadata &&
+        previousMetadata.is_unlisted &&
+        updatedMetadata.is_unlisted === false
+      ) {
+        // Mark track as publishing
+        updatedMetadata._is_publishing = true
+      }
+
+      // Add premium metadata if needed
+      const userWallet = currentUser?.erc_wallet || currentUser?.wallet
+      if (userWallet && isUsdcPurchaseEnabled) {
+        updatedMetadata = await addPremiumMetadata(
+          updatedMetadata,
+          userWallet,
+          userBank
+        )
+      }
+
+      // Format musical key
+      if (updatedMetadata.musical_key) {
+        updatedMetadata.musical_key =
+          formatMusicalKey(updatedMetadata.musical_key) ?? null
+
+        // Set custom musical key flag if it's changed or was already set
+        if (previousMetadata) {
+          updatedMetadata.is_custom_musical_key =
+            previousMetadata.is_custom_musical_key ||
+            (!!updatedMetadata.musical_key &&
+              updatedMetadata.musical_key !== previousMetadata.musical_key)
+        }
+      }
+
+      // Format BPM
+      if (updatedMetadata.bpm) {
+        updatedMetadata.bpm = Number(updatedMetadata.bpm)
+
+        // Set custom BPM flag if it's changed or was already set
+        if (previousMetadata) {
+          updatedMetadata.is_custom_bpm =
+            previousMetadata.is_custom_bpm ||
+            (!!updatedMetadata.bpm &&
+              updatedMetadata.bpm !== previousMetadata.bpm)
+        }
+      }
+
       const sdkMetadata = trackMetadataForUploadToSdk(
-        metadata as TrackMetadataForUpload
+        updatedMetadata as TrackMetadataForUpload
       )
 
+      // Determine if we need to generate a preview
```

Comment:
nit: this logic is kinda dense, an extra comment here would help I think!

```suggestion
      // Determine if we need to generate a preview - check if preview_start has changed OR track file has changed
```